src/runtime/HalideBuffer.h

```diff
@@ -1124,6 +1124,19 @@ class Buffer {
         return dst;
     }
 
+    /** Make a copy of the Buffer which shares the underlying host and/or device
+     * allocations as the existing Buffer. This is purely syntactic sugar for
+     * cases where you have a const reference to a Buffer but need a temporary
+     * non-const copy (e.g. to make a call into AOT-generated Halide code), and want a terse
+     * inline way to create a temporary. \code
+     * void call_my_func(const Buffer<const uint8_t>& input) {
+     *     my_func(input.alias(), output);
+     * }\endcode
+     */
+    inline Buffer<T, D> alias() const {
+        return *this;
```

Comment:
I was thinking you'd want to return `Buffer<T, D>(buf)` or similar - a non-owning view of the same host/dev allocation. Returning `this` increments the ref count. Maybe that overhead is worth the additional safety of not faulting if someone keeps the alias alive beyond the lifetime of the original and the additional simplicity of it just invoking the copy constructor.

Reply:
I will bet you `${HUGE_SUM}` that if we provide this function, someone will indeed misuse it as you describe.

Reply:
Expanding a bit more on this, seems to me like the options are:
- have `alias()` return a full copy. Safest but slowest.
- have `alias()` return a non-owning view. This is a bit more efficient, but runs the risk of misuse of the sort that may or may not be immediately detected unless running in MSAN/ASAN builds.
- have `alias()` (somehow) return a non-owning `halide_buffer_t` that can somehow only be used for the purpose of calling into an AOT function. We could do this by defining custom structs that accept either `halide_buffer_t*` or an opaque struct value returned by `alias()`... but it smells a little like premature optimization. 

I'm somewhat inclined to go with the safe-but-slow option and then consider the final sneaky option if we get evidence that it's a performance issue.